lib/guardian/post_guardian.rb

```diff
@@ -100,8 +100,9 @@ def can_delete_all_posts?(user)
     is_staff? &&
     user &&
     !user.admin? &&
-    (user.first_post_created_at.nil? || user.first_post_created_at >= SiteSetting.delete_user_max_post_age.days.ago) &&
-    user.post_count <= SiteSetting.delete_all_posts_max.to_i
+    (is_admin? ||
```

Comment:
The `is_admin?` condition doesn't look right to me. There is a `!user.admin?` check before so `is_admin?` will never be `true`